src/ErrorReporting/Bootstrap.php

```diff
@@ -0,0 +1,212 @@
+                    'batchOptions' => ['workerNum' => 2]
+                ]
+            );
+        register_shutdown_function(
+            [
+                '\\Google\\Cloud\\ErrorReporting\\Bootstrap',
+                'shutdownHandler'
+            ]
+        );
+        set_exception_handler(
+            array(
+                '\\Google\\Cloud\\ErrorReporting\\Bootstrap',
+                'exceptionHandler'
+            )
+        );
+        set_error_handler(
+            array(
+                '\\Google\\Cloud\\ErrorReporting\\Bootstrap',
+                'errorHandler'
+            )
+        );
+    }
+
+    /**
+     * Return a string prefix for the given error level.
+     *
+     * @param int $level
+     * @return string A string prefix for reporting the error.
+     */
+    public static function getErrorPrefix($level)
+    {
+        switch ($level) {
+            case E_PARSE:
+                $prefix = 'PHP Parse error';
+                break;
+            case E_ERROR:
+            case E_CORE_ERROR:
+            case E_COMPILE_ERROR:
+                $prefix = 'PHP Fatal error';
+                break;
+            case E_USER_ERROR:
+            case E_RECOVERABLE_ERROR:
+                $prefix = 'PHP error';
+                break;
+            case E_WARNING:
+            case E_CORE_WARNING:
+            case E_COMPILE_WARNING:
+            case E_USER_WARNING:
+                $prefix = 'PHP Warning';
+                break;
+            case E_NOTICE:
+            case E_USER_NOTICE:
+                $prefix = 'PHP Notice';
+                break;
+            case E_STRICT:
+                $prefix = 'PHP Debug';
+                break;
+            default:
+                $prefix = 'PHP Notice';
+        }
+        return $prefix;
+    }
+
+    /**
+     * Return an error level string for the given PHP error level.
+     *
+     * @param int $level
+     * @return string An error level string.
+     */
+    public static function getErrorLevelString($level)
+    {
+        switch ($level) {
+            case E_PARSE:
+                return 'CRITICAL';
+            case E_ERROR:
+            case E_CORE_ERROR:
+            case E_COMPILE_ERROR:
+            case E_USER_ERROR:
+            case E_RECOVERABLE_ERROR:
+                return 'ERROR';
+            case E_WARNING:
+            case E_CORE_WARNING:
+            case E_COMPILE_WARNING:
+            case E_USER_WARNING:
+                return 'WARNING';
+            case E_NOTICE:
+            case E_USER_NOTICE:
+                return 'NOTICE';
+            case E_STRICT:
+                return 'DEBUG';
+            default:
+                return 'NOTICE';
+        }
+        return $prefix;
+    }
+
+    /**
+     * @param mixed $ex \Throwable (PHP 7) or \Exception (PHP 5)
+     */
+    public static function exceptionHandler($ex)
+    {
+        $message = sprintf('PHP Notice: %s', (string)$ex);
+        self::$psrBatchLogger->error($message);
```

Comment:
There's no guarantee that `self::init()` will have been called before this method is invoked, so you should check that `self::$psrBatchLogger` is set.